### ISSUE TITLE
cl.BaseURL

### ISSUE THREAD

**Issue by tmorrell** [caltechlibrary/cl-js#3]
I was doing some testing and wanted to set the url to our S3 test bucket

cl.BaseURL gets set by the widget here: https://github.com/caltechlibrary/cl-js/blob/9cff436c886514d9af4834cf3314b153b168d6ec/scripts/CL-BuilderWidget.js#L371

but it doesn't do anything. I think the variable is actually cl.FeedsBaseURL (https://github.com/caltechlibrary/cl-js/blob/9cff436c886514d9af4834cf3314b153b168d6ec/scripts/CL-feeds.js#L40), but since it is hard coded in the function there doesn't seem to be a way to modify it without editing CL.js.

It's not that bad to modify CL.js for testing, and I think we can just delete the portion of the widget code that sets cl.BaseURL.  We could also create a method to update the base url (or whatever would make sense for javascript).

**Comment by rsdoiel** [comment]
I think originally (the widget has evolved a bit) I was using the two different names to make a distinction between runtime code and a global default value. CL.FeedsBaseURL is the canonical value if none is set.  Overtime I think I've moved away from needing to know a runtime cl.BaseURL value (it was used to determine where to fetch the JSON lists for populating both the widget builder and the widget output). At this stage it's probably anachronistic.
Feeds is the authoritative source for the widget interaction and I don't think anyone uses the parts of the CL-js libraries that make CL.js. It's just not that big.

**Comment by rsdoiel** [comment]
Another thought.  When I do development on CL.js I test by parts (that's something I highly recommend) but in the production pages I used the assembled CL.js so everything is tied to a common version of assumptions. :-)

**Comment by tmorrell** [comment]
Makes sense. I think we can just pull cl.BaseURL from the widget output so folks don't get confused. I've made the change in main CL-BuilderWidget.js, but you should probably test before putting into production.

**Comment by rsdoiel** [comment]
Sounds, good, I'll test and then push the changes into the main branch of
feeds.library.caltech.edu repo.

thanks!

Robert

**Comment by rsdoiel** [comment]
I was reviewing the test code (see scripts/test-*.md in the repo). The cl.BaseURL was used to have different feeds available on the same page. It really shouldn't be called "BaseURL" but rather "FeedURL". I am not 100% sure (until I finish reading through and testing the code) but it may not actually get used.  I'm still poking at things but if it is NOT used then I'll remove it.

**Comment by rsdoiel** [comment]
OK, I've read through the code and I remember what I was trying to do about BaseURL. The code is built of generally in CL-core.js.  It isn't specific to feeds (we could build other things with it as an example).  BaseURL is used at this level so you have something you can depend on for knowing where your code came from.  In the CL-feeds.js I created a FeedsBaseURL to be explicit about what I am considering the working base.  I don't think the widget needs to generate code that sets BaseURL since if they are using CL.js as the widget suggests they will never know about that attribute in the core object. I'm going to leave BaseURL in the core object. What I am going to think about is whether or not it makes sense to rename FeedsBaseURL and just use BaseURL through out to avoid the confusion. If feeds data was on another machine where we were setting things deeper in the tree having two attributes could be useful but I seriously doubt that use case.  I'm going to leave this ticket open for the time being while I mull this over.
